src/compiler/builder.ts

```diff
@@ -209,7 +211,7 @@ namespace ts {
             state.cleanedDiagnosticsOfLibFiles = true;
             const options = state.program.getCompilerOptions();
             if (forEach(state.program.getSourceFiles(), f =>

```

Comment:
I still don't like looping over every file and then testing if it's the default library -- we should be able to get that in O(1).

Reply:
There are multiple default libvraries so we do need to go over them and its only once.

Reply:
Note this doesn't happen everytime. It happens only once in build cycle. not on every affected file. (its guarded by state.cleanedDiagnosticsOfLibFiles

Reply:
OK.